lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -179,18 +191,34 @@ def url_download( self, install_dir, downloaded_file_name, download_url, extract
                 src.close()
             if dst:
                 dst.close()
-        
-        #try:
-        if checksums.has_key('sha256sum'):
+
+        if checksums.has_key('sha256sum') or '#sha256#' in download_url:
             downloaded_checksum = hashlib.sha256(open(file_path, 'rb').read()).hexdigest()
-            if downloaded_checksum != checksums['sha256sum']:
-                raise Exception( 'Given sha256 checksum does not match with the one from the downloaded file (%s).' % (downloaded_checksum) )
-        
-        if checksums.has_key('md5sum'):
+
+            # Determine expected value
+            if checksums.has_key('sha256sum'):
+                expected = checksums['sha256sum']
+            else:
+                expected = download_url.split('#sha256#')[1]
+
+            if downloaded_checksum != expected:
```

Comment:
`hashlib.hexdigest()` docs do not state whether implementations should return lowercase, although at least for CPython this appears to be the case.

It's probably more likely that `expected` would have uppercase characters. However, it's probably safest to just `.lower()` everything.